wundergraph_cli/src/print_schema/mod.rs

```diff
@@ -311,41 +312,51 @@ mod tests {
         std::thread::sleep(std::time::Duration::from_secs(1));
 
         let query = "{\"query\": \"{ Users { id  name  } } \"}";
-        let mut r = client
-            .post(&format!("http://{}/graphql", listen_url))
-            .body(query)
-            .header(
-                reqwest::header::CONTENT_TYPE,
-                reqwest::header::HeaderValue::from_static("application/json"),
-            )
-            .send()
-            .unwrap();
-        insta::assert_json_snapshot!(r.json::<serde_json::Value>().unwrap());
-
         let mutation = r#"{"query":"mutation CreateUser {\n  CreateUser(NewUser: {name: \"Max\"}) {\n    id\n    name\n  }\n}","variables":null,"operationName":"CreateUser"}"#;
+        let t1 = request_test(&client, &listen_url, query);
+        let t2 = request_test(&client, &listen_url, mutation);
+        let t3 = request_test(&client, &listen_url, query);
+
+        child.kill().unwrap();
+        child.wait().unwrap();
+
+        if !t1 || !t2 || !t3 {
+            panic!("round_trip failed")
+        }
+    }
+
+    fn request_test(
+        client: &reqwest::Client,
+        listen_url: &str,
+        body: &'static str,
+    ) -> bool {
         let mut r = client
```

Comment:
I think that request should be also executed using this separated thread below as we use `unwrap()` here.

Reply:
I've had some difficulty to move request into swap. `rustc` complained "borrowed value does not live long enough". Therefore 375c03aea376cb057d3261145ee1344ed076fba5 abandons `unwrap()` in this case. But I'm eager to now how request could be moved into `swap`.